src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -3148,5 +3181,578 @@ public C1(string arg)
             CompileAndVerify(comp, expectedOutput: "{1, qq1}");
         }
 
+        [Fact]
+        public void Inference01()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        Test((null, null));
+        Test((1, 1));
```

Comment:
from 1 we can infer   int, and then we have exact match in the overload resolution